lib/simplecov/raw_coverage.rb

```diff
@@ -0,0 +1,38 @@
+module SimpleCov
+  module RawCoverage
+  module_function
+
+    # Merges multiple Coverage.result hashes
+    def merge_results(*results)
+      results.reduce({}) do |result, merged|
+        merge_resultsets(result, merged)
+      end
+    end
+
+    # Merges two Coverage.result hashes
+    def merge_resultsets(result1, result2)
+      (result1.keys | result2.keys).each_with_object({}) do |filename, merged|
+        file1 = result1[filename]
+        file2 = result2[filename]
+        merged[filename] = merge_file_coverage(file1, file2)
+      end
+    end
+
+    def merge_file_coverage(file1, file2)
+      return (file1 || file2).dup unless file1 && file2
+
+      file1.zip(file2).map do |count1, count2|
```

Comment:
Whoops, that was so lazy that it didn't even create a result array. A correct implementation (with a `.to_a` on the end) is slower:

```
Warming up --------------------------------------
   lazy.zip.map.to_a     1.450k i/100ms
             zip.map     7.869k i/100ms
      map.with_index    10.748k i/100ms
     each_with_index     9.976k i/100ms
Calculating -------------------------------------
   lazy.zip.map.to_a     13.688k (±14.1%) i/s -     68.150k in   5.097768s
             zip.map     79.369k (± 8.6%) i/s -    393.450k in   5.005312s
      map.with_index    110.505k (± 4.0%) i/s -    558.896k in   5.067126s
     each_with_index    114.300k (± 4.2%) i/s -    578.608k in   5.071720s

Comparison:
     each_with_index:   114300.5 i/s
      map.with_index:   110505.2 i/s - same-ish: difference falls within error
             zip.map:    79368.7 i/s - 1.44x  slower
   lazy.zip.map.to_a:    13688.2 i/s - 8.35x  slower
```